LTS release checklist (#122)
jenkins-infra/release: .github/ISSUE_TEMPLATE/1-lts-release-checklist.md

---

> name: "🥇 New LTS release checklist"

Reviewer: Why 🥇?

Author: Random emoji I picked, can remove or replace with a better one

labels: lts
about: Track work required for a new LTS release
---

# Next LTS release

## Release Lead

<!-- 
The release lead is the person who makes sure that all steps are completed
Not necessarily the person doing all the work

This role should rotate between LTS releases
-->

> @<github-username of release lead>

Reviewer 2: I love this idea, to know who's responsible

Reviewer 3: Is it worth having a separate assignment of someone that monitors the results of the build?

As an example, I've recently been monitoring LTS and weekly release execution with Oliver as the release lead.  He created the initial pull request, walked it through the review process, and merged it.  He then relied on others to confirm the final build and packaging process was successful.  

Would that work or is the assumption that the release lead watches things all the way to completion of the release?

Author: I had the release lead as the person who was making sure everything got done, not necessarily doing the things, anyone can put their name next to a task / put their hand up to do part of the release

## Prep work

> - [ ] LTS baseline selected

Reviewer 2: Would it makes sense to have a link to the discussion happening in the ML?

Author: Yes this is missing the pre selection step of looking for consensus on a baseline

> - [ ] LTS baseline selected

Reviewer 3: ```suggestion
- [ ] LTS baseline discussed and selected in the [Jenkin developers mailing list](https://groups.google.com/g/jenkinsci-dev)
```

> - [ ] LTS branch created in jenkinsci/jenkins, e.g. `stable-2.263` (strike this out for new point release)

Reviewer 3: Group branch creation together in the checklist?

```suggestion
- [ ] Create or update release branch in [jenkinsci/jenkins](https://github.com/jenkinsci/jenkins), e.g. `stable-2.263`

- [ ] Create or update release branch in [jenkins-infra/release](https://github.com/jenkins-infra/release), e.g. `stable-2.263`

- [ ] Create or update release branch in [jenkinsci/packaging](https://github.com/jenkinsci/packaging), e.g. `stable-2.263`
```

- [ ] Create pull request to update bom to new release line (strike this out for new point release)

- [ ] Create pull request to update configuration-as-code integration tests to new release line (strike this out for new point release)

- [ ] Backporting announcement email

- [ ] Review Jira and GitHub pull requests for additional candidates

- [ ] Open backporting PR with into-lts label and summary of changes in description from script

- [ ] Review ATH, bom and configuration-as-code integration tests results

> - [ ] Prepare LTS changelog

Reviewer 3: ```suggestion
- [ ] Prepare [LTS changelog](https://www.jenkins.io/changelog-stable/) based on the [style guide](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/_STYLEGUIDE.adoc) using the [changelog generator](https://github.com/jenkinsci/core-changelog-generator/blob/master/README.md)
```

> - [ ] Prepare LTS upgrade guide

Reviewer 3: ```suggestion
- [ ] Prepare [LTS upgrade guide](https://www.jenkins.io/doc/upgrade-guide/) based on [previous upgrade guides](https://github.com/jenkins-infra/jenkins.io/tree/master/content/_data/upgrades)
```

## RC creation

> - [ ] Merge backporting PR in jenkinci/jenkins

Reviewer: ```suggestion
- [ ] Merge backporting PR in jenkinci/jenkins using a merge commit (do not squash)
```

> - [ ] Create or update release branch in jenkins-infra/release, e.g. `rc-stable-2.263`.

Reviewer 3: ```suggestion
- [ ] Create or update release branch in [jenkins-infra/release](https://github.com/jenkins-infra/release), e.g. `rc-stable-2.263`.
```

> - [ ] Create or update packaging branch in jenkinsci/packaging, e.g. `stable-2.263`

Reviewer 3: ```suggestion
- [ ] Create or update packaging branch in [jenkinsci/packaging]([jenkinsci/packaging](https://github.com/jenkinsci/packaging)), e.g. `stable-2.263`
```

- [ ] Run job on [release.ci.jenkins.io](https://release.ci.jenkins.io/blue/organizations/jenkins/core%2Fstable%2Frelease/branches/) # TODO update url

- [ ] Publish pre-release Github release

- [ ] Send announcement email

> 

Reviewer: Add here: Check with security team that no security update is planned. If there is, the rest of this checklist does not apply (but hold any updates until the public pre-announcement to the jenkinsci-advisories mailing list).

While every effort is done to inform release team, better safe than sorry. Especially with rotating people.

Author: What do you mean by the rest of the checklist does not apply?

Doesn't the changelog still need publishing, along with all the other checks done, and someone needs to run the release albeit a different job?

## LTS release

- [ ] Check LTS changelog status

> - [ ] Create or update release branch in jenkins-infra/release, e.g. `stable-2.263`

Reviewer: Should happen way before then to allow for review.

Reviewer 3: > Should happen way before then to allow for review.

I believe this checklist item is to confirm the changelog is visible on jenkins.io after the release is complete.  Creation and review of the changelog and the upgrade guide are different entries in the checklist.

- [ ] Run job on [release.ci.jenkins.io](https://release.ci.jenkins.io/blue/organizations/jenkins/core%2Fstable%2Frelease/branches/)

> 

Reviewer 2: Probably missing a link to the monitoring dashboard

Reviewer 3: Added.  Thanks for the reminder.

> - [ ] Publish changelog

Reviewer: Needs to happen at least 1 day before release if it's a security update to give us time to rebase any security content.

---
name: "🥇 New LTS release checklist"
labels: lts
about: Track work required for a new LTS release
---

# Next LTS release

## Release Lead

<!-- 
The release lead is the person who makes sure that all steps are completed
Not necessarily the person doing all the work

This role should rotate between LTS releases
-->

@<github-username of release lead>

## Prep work

- [ ] LTS baseline discussed and selected in the [Jenkins developers mailing list](https://groups.google.com/g/jenkinsci-dev)

- [ ] Create or update release branch in [jenkinsci/jenkins](https://github.com/jenkinsci/jenkins), e.g. `stable-2.263`

- [ ] Create or update release branch in [jenkins-infra/release](https://github.com/jenkins-infra/release), e.g. `stable-2.263`

- [ ] Create or update release branch in [jenkinsci/packaging](https://github.com/jenkinsci/packaging), e.g. `stable-2.263`

> - [ ] Create pull request to update bom to new release line (strike this out for new point release)

Reviewer 3: ```suggestion
- [ ] Create pull request to update [bom](https://github.com/jenkinsci/bom) to new release line (strike this out for new point release)
```

- [ ] Prepare [LTS upgrade guide](https://www.jenkins.io/doc/upgrade-guide/) based on [previous upgrade guides](https://github.com/jenkins-infra/jenkins.io/tree/master/content/_data/upgrades)

## RC creation

- [ ] Merge backporting PR in jenkinci/jenkins using a merge commit (do not squash)

- [ ] Create or update release branch in [jenkins-infra/release](https://github.com/jenkins-infra/release), e.g. `rc-stable-2.263`.

- [ ] Create or update packaging branch in [jenkinsci/packaging]([jenkinsci/packaging](https://github.com/jenkinsci/packaging)), e.g. `stable-2.263`

- [ ] Run job on [release.ci.jenkins.io](https://release.ci.jenkins.io/job/core/job/stable-rc)

- [ ] Publish [Github release](https://github.com/jenkinsci/jenkins/releases) using the GitHub changelog draft

- [ ] Send announcement email

- [ ] Check with security team that no security update is planned.  If a security update is planned, revise the checklist after the public pre-announcement to the [jenkinsci-advisories mailing list](https://groups.google.com/g/jenkinsci-advisories)

> - [ ] Create draft changelog and draft upgrade guide as a jenkins.io pull request

Reviewer 3: Apologies for the duplication.  Never done a code review of a checklist before.  Apparently not great at it yet.
```suggestion
```

## LTS release

- [ ] Check [LTS changelog](https://www.jenkins.io/changelog-stable/) status

> - [ ] Create or update release branch in jenkins-infra/release, e.g. `stable-2.263`

Reviewer 3: Already listed earlier in the checklist before the release candidate
```suggestion
```

> - [ ] Create or update release branch in jenkins-infra/release, e.g. `stable-2.263`

Reviewer 3: This is already covered in an earlier checklist item
```suggestion
```

- [ ] Run job on [release.ci.jenkins.io](https://release.ci.jenkins.io/blue/organizations/jenkins/core%2Fstable%2Frelease/branches/)

- [ ] Publish changelog (one day prior to the release in case of a security update)

> - [ ] Publish GitHub release pointing to LTS changelog

Reviewer 3: ```suggestion
- [ ] Publish [GitHub release](https://github.com/jenkinsci/jenkins/releases) pointing to LTS changelog

- [ ] Confirm [Datadog checks](https://p.datadoghq.com/sb/0Igb9a-e6849e5e019250ef5aaea3589297fe8b) are passing

- [ ] Confirm the [Debian installer acceptance test](https://ci.jenkins.io/job/Infra/job/acceptance-tests/job/install-lts-debian-package/) is passing

- [ ] Confirm the [Red Hat installer acceptance test](https://ci.jenkins.io/job/Infra/job/acceptance-tests/job/install-lts-redhat-rpm/) is passing

- [ ] Adjust state of all [Jira issues](https://issues.jenkins.io/) fixed in the release (see the [changelog](https://www.jenkins.io/changelog-stable) for issue links)
```

- [ ] Backporting announcement email

- [ ] Review Jira and GitHub pull requests for additional candidates

- [ ] Open backporting PR with into-lts label and summary of changes in description from script

- [ ] Review ATH, bom and configuration-as-code integration tests results

- [ ] Prepare [LTS changelog](https://www.jenkins.io/changelog-stable/) based on the [style guide](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/_STYLEGUIDE.adoc) using the [changelog generator](https://github.com/jenkinsci/core-changelog-generator/blob/master/README.md)

- [ ] Prepare [LTS upgrade guide](https://www.jenkins.io/doc/upgrade-guide/) based on [previous upgrade guides](https://github.com/jenkins-infra/jenkins.io/tree/master/content/_data/upgrades)

## RC creation

- [ ] Merge backporting PR in jenkinci/jenkins using a merge commit (do not squash)

- [ ] Create or update release branch in [jenkins-infra/release](https://github.com/jenkins-infra/release), e.g. `rc-stable-2.263`.

- [ ] Create or update packaging branch in [jenkinsci/packaging]([jenkinsci/packaging](https://github.com/jenkinsci/packaging)), e.g. `stable-2.263`

> - [ ] Run job on [release.ci.jenkins.io](https://release.ci.jenkins.io/job/Core/) # TODO update url

Reviewer 3: Confirmed URL on release.ci.jenkins.io
```suggestion
- [ ] Run job on [release.ci.jenkins.io](https://release.ci.jenkins.io/job/core/job/stable-rc)
```

- [ ] Publish [Github release](https://github.com/jenkinsci/jenkins/releases) using the GitHub changelog draft

- [ ] Send announcement email

> 

Reviewer 3: ```suggestion

- [ ] Check with security team that no security update is planned.  If a security update is planned, revise the checklist after the public pre-announcement to the [jenkinsci-advisories mailing list](https://groups.google.com/g/jenkinsci-advisories)

- [ ] Create draft changelog and draft upgrade guide as a jenkins.io pull request
```

## LTS release

- [ ] Check [LTS changelog](https://www.jenkins.io/changelog-stable/) status

- [ ] Create or update release branch in jenkins-infra/release, e.g. `stable-2.263`

- [ ] Run job on [release.ci.jenkins.io](https://release.ci.jenkins.io/blue/organizations/jenkins/core%2Fstable%2Frelease/branches/)

> - [ ] Publish changelog

Reviewer 3: ```suggestion
- [ ] Publish changelog (one day prior to the release in case of a security update)
```

- [ ] Publish GitHub release pointing to LTS changelog

> <!--
> Put an `x` into the [ ] to show you have filled the information below
> Describe your issue below
> -->

Author: ```suggestion
```

- [ ] Prepare [LTS upgrade guide](https://www.jenkins.io/doc/upgrade-guide/) based on [previous upgrade guides](https://github.com/jenkins-infra/jenkins.io/tree/master/content/_data/upgrades)

## RC creation

- [ ] Merge backporting PR in jenkinci/jenkins using a merge commit (do not squash)

- [ ] Create or update release branch in [jenkins-infra/release](https://github.com/jenkins-infra/release), e.g. `rc-stable-2.263`.

- [ ] Create or update packaging branch in [jenkinsci/packaging]([jenkinsci/packaging](https://github.com/jenkinsci/packaging)), e.g. `stable-2.263`

- [ ] Run job on [release.ci.jenkins.io](https://release.ci.jenkins.io/job/core/job/stable-rc)

- [ ] Publish [Github release](https://github.com/jenkinsci/jenkins/releases) using the GitHub changelog draft

- [ ] Send announcement email

- [ ] Check with security team that no security update is planned.  If a security update is planned, revise the checklist after the public pre-announcement to the [jenkinsci-advisories mailing list](https://groups.google.com/g/jenkinsci-advisories)

## LTS release

> - [ ] Check [LTS changelog](https://www.jenkins.io/changelog-stable/) status

Reviewer 3: ```suggestion
- [ ] Check [LTS changelog](https://www.jenkins.io/changelog-stable/) is visible on the downloads site
```

- [ ] Run job on [release.ci.jenkins.io](https://release.ci.jenkins.io/blue/organizations/jenkins/core%2Fstable%2Frelease/branches/)

- [ ] Publish changelog (one day prior to the release in case of a security update)

- [ ] Publish [GitHub release](https://github.com/jenkinsci/jenkins/releases) pointing to LTS changelog

- [ ] Confirm [Datadog checks](https://p.datadoghq.com/sb/0Igb9a-e6849e5e019250ef5aaea3589297fe8b) are passing

- [ ] Confirm the [Debian installer acceptance test](https://ci.jenkins.io/job/Infra/job/acceptance-tests/job/install-lts-debian-package/) is passing

- [ ] Confirm the [Red Hat installer acceptance test](https://ci.jenkins.io/job/Infra/job/acceptance-tests/job/install-lts-redhat-rpm/) is passing

> - [ ] Adjust state of all [Jira issues](https://issues.jenkins.io/) fixed in the release (see the [changelog](https://www.jenkins.io/changelog-stable) for issue links)

Author: what does this mean @MarkEWaite? update the released as field?
